specs/core/0_beacon-chain.md

```diff
@@ -1554,11 +1571,9 @@ def process_final_updates(state: BeaconState) -> None:
             validator.effective_balance = min(balance - balance % EFFECTIVE_BALANCE_INCREMENT, MAX_EFFECTIVE_BALANCE)
     # Update start shard
     state.latest_start_shard = (state.latest_start_shard + get_shard_delta(state, current_epoch)) % SHARD_COUNT
-    # Set active index root
-    index_root_position = (next_epoch + ACTIVATION_EXIT_DELAY) % LATEST_ACTIVE_INDEX_ROOTS_LENGTH
-    state.latest_active_index_roots[index_root_position] = hash_tree_root(
-        get_active_validator_indices(state, next_epoch + ACTIVATION_EXIT_DELAY)
-    )
+    # Set active registry root
+    active_registry_index = (next_epoch + ACTIVATION_EXIT_DELAY) % LATEST_ACTIVE_REGISTRY_ROOTS_LENGTH
+    state.latest_active_registry_roots[active_registry_index] = get_active_registry_root(state)
```

Comment:
This usage is incorrect. You are inserting the _current_ epoch active registry into the epoch position `ACTIVATION_EXIT_DELAY` in the future.

This is purposefully supposed to be populating the future epoch position with the future registry.